python/pyspark/pandas/data_type_ops/boolean_ops.py

```diff
@@ -73,7 +77,9 @@ def sub(self, left, right) -> Union["Series", "Index"]:
             return left - right
 
     def mul(self, left, right) -> Union["Series", "Index"]:
-        if not is_valid_operand_for_numeric_arithmetic(right, allow_bool=False):
+        if not is_valid_operand_for_numeric_arithmetic(
+            right, allow_bool_index_ops=False, allow_bool=False
```

Comment:
This should allow bools and behave like `&`?

```py
>>> pd.Series([True, False, True]) * pd.Series([True, False, False])
0     True
1    False
2    False
dtype: bool
>>> pd.Series([True, False, True]) * True
0     True
1    False
2     True
dtype: bool
>>> pd.Series([True, False, True]) * False
0    False
1    False
2    False
dtype: bool
```

Reply:
Good catch! `pd.Series([True, False, True]) * True` and `pd.Series([True, False, True]) * False` are supported.

Reply:
+/* between bool Series/Index are supported now.